lgc/builder/YCbCrConverter.cpp

```diff
@@ -326,61 +328,119 @@ void YCbCrConverter::genSamplerDescChroma() {
 // =====================================================================================================================
 // Generate image descriptor for chroma channel
 void YCbCrConverter::genImgDescChroma() {
-  Value *pInt32One = ConstantInt::get(m_builder->getInt32Ty(), 1);
   SqImgRsrcRegHandler proxySqRsrcRegHelper(m_builder, m_imgDescLuma, m_gfxIp);
-  YCbCrAddressHandler addrHelper(m_builder, &proxySqRsrcRegHelper, m_gfxIp);
-  unsigned bpp = (m_metaData.word1.planes == 1)
-                     ? (m_metaData.word2.bitCounts.xBitCount + m_metaData.word2.bitCounts.yBitCount +
-                        m_metaData.word2.bitCounts.zBitCount + m_metaData.word2.bitCounts.wBitCount)
-                     : m_metaData.word2.bitCounts.xBitCount;
-  addrHelper.genHeightAndPitch(m_metaData.word0.bitDepth.channelBitsR, bpp, m_metaData.word2.bitCounts.xBitCount,
-                               m_metaData.word1.tileOptimal, m_metaData.word1.planes);
-  addrHelper.genBaseAddress(m_metaData.word1.planes);
-
   Value *width = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Width);
   Value *height = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Height);
-  Value *widthHalf = m_builder->CreateLShr(width, pInt32One);
-  Value *heightHalf = m_builder->CreateLShr(height, pInt32One);
   m_width = m_builder->CreateUIToFP(width, m_builder->getFloatTy());
   m_height = m_builder->CreateUIToFP(height, m_builder->getFloatTy());
 
-  Value *pPitch = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Pitch);
-  Value *pitchHalf = m_builder->CreateLShr(pPitch, pInt32One);
+  if (m_metaData.word1.planes > 1) {
```

Comment:
why do we need so complex logic to get the address of second and third plane? could you explain more? thanks!

Reply:
These logic are used for tracing back for the first plane's resource address, it might be not necessary, do you have some suggestions?

Reply:
> why do we need so complex logic to get the address of second and third plane? could you explain more? thanks!

This logic is somehow like the code in ImageBuilder::CreateImageSample which is  implemented by Tim, I'm not sure if there's a simple method.

Reply:
The code in CreateImageSample to find the descriptor load is an optimization (to try and include the descriptor load in the waterfall loop). I don't think we can rely on it always succeeding in finding the descriptor load.

Reply:
> The code in CreateImageSample to find the descriptor load is an optimization (to try and include the descriptor load in the waterfall loop). I don't think we can rely on it always succeeding in finding the descriptor load.

I haven't foreseen under which condition it will fail to find the descriptor load.(If could, could you please supply some case). I'd thankful if you or Nicolai could provide a solution to cover the cases listed in #760

Reply:
Oh sorry, I was getting confused with the waterfall case.

Yes, we already have code in CreateImageSample that finds the descriptor load to see if it is a converting sampled image. I guess that works in all the cases that you have seen, although we know there is a theoretical problem because of the way the extension was defined with no spir-v changes.